src/funcs.c

```diff
@@ -2795,4 +2824,111 @@ void joy_right()
 
   set(app->LV_GamesList, MUIA_List_Active, ind-1);
 
-}
\ No newline at end of file
+}
+
+
+void add_single_whdload()
```

Comment:
A lot of the code I see here looks similar with the follow_thread(). Instead of replicating the same code, maybe it would be better to split it to different methods and refactor the code to use these. Otherwise it will become difficult to maintain this code.

Reply:
yes. Its based in follow_thread() but didnt want to make  changes to other people's code but I think you are alright.